backend/src/search/views.py

```diff
@@ -22,104 +34,74 @@ async def root() -> dict:
 @router.post(
     "/search/file/{user_id}", status_code=200, dependencies=[Depends(is_authorized)]
 )
-async def create_search_task(file: UploadFile, user_id: int) -> dict:
+async def create_search_task(
+    file: UploadFile, user_id: int, db_session: Session = Depends(get_db_session)
+) -> SearchTaskCreationResponseModel:
     if file.content_type != "application/pdf":
         raise HTTPException(400, detail="Invalid document type.")
-
     try:
-        file_content: BinaryIO = file.file
+        file_contents = file.file.read()
+        search = SearchRepository.create_search(
+            db_session=db_session,
+            user_id=user_id,
+            file_name=file.filename,
+        )
+        task_result = celery.send_task("search", (file_contents, search.id))
+        SearchRepository.update(db_session, search, {"task_id": task_result.id})
     except Exception:
         raise HTTPException(500, detail="Internal server error.")
-
-    return {"message": "successful upload", "filename": file.filename}
+    return SearchTaskCreationResponseModel(filename=file.filename)
```

Comment:
Please save in the database that this search failed with status ERROR, the same should be implemented for worker if exception happends